plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1049,6 +1050,9 @@ public Optional<ConnectorOutputMetadata> finishCreateTable(ConnectorSession sess
             // May happen when table has been already created concurrently.
             throw new TrinoException(TABLE_ALREADY_EXISTS, format("Table %s already exists", icebergTableHandle.name()), e);
         }
+        catch (RuntimeException e) {
```

Comment:
> Classify Iceberg IO errors as external

The commit title is misleading. The exception isn't limited to IO errors. 

> for example with a permission error

Please share the stacktrace to PR description.

Reply:
I updated the commit message, PTAL